torch/nn/modules/module.py

```diff
@@ -90,6 +95,78 @@ def __setstate__(self, state: Dict):
 _EXTRA_STATE_KEY_SUFFIX = '_extra_state'
 
 
+def register_buffer_registration_hook(hook: Callable[..., None]) -> RemovableHandle:
+    r"""Registers a buffer registration hook common to all modules.
+
+    .. warning ::
+
+        This adds global state to the `nn.Module` module
+
+    The hook will be called every time :func:`register_buffer` is invoked.
+    It should have the following signature::
+
+        hook(name, module) -> None or modified input
```

Comment:
Shouldn't the buffer be passed in?
Also for consistency, could you pass the module first: `hook(module, name, buffer) -> None or new buffer`

Reply:
Same for the others below

Reply:
yep, good catch. That's my bad for just copying the docstring over and not replacing all instances 😅